source/class/qxl/apiviewer/ui/SearchView.js

```diff
@@ -384,6 +384,7 @@ qx.Class.define("qxl.apiviewer.ui.SearchView", {
 
       sresult = this._searchIndex(search[0], search[1]);
       sresult.sort(this._sortByIcons);
+      sresult = this.__withoutDuplicates(sresult);
```

Comment:
This looks like it's covering the symptom rather than fixing the cause. What is causing the duplicates? Shouldn't that be fixed?

Reply:
@derrell you are right. Will check it later more deeply.

Reply:
@derrell diving to the problem I discover another one. When you using autosearch there are packages in list instead of classes. Now about duplicating problem there are different types of nodes. Package, class, method etc. I checked resulting internal tree and was surprised there are no packages or nodes with this type. But classes and packages have the same type of node and it is a package type. Also still havent checked out usual tree which built for usual search by opening nodes but there everything is fine. Seems like different processes. In conclusion, I d say the problem is wrong type for node. For example I type qx.ui.ba and got qx.ui.basic node which is package now and has 3 children also references to packages.

Reply:
great that the api viewer is getting some love ... thank you @goldim for digging in!